notebooks/ir_ima_visualization/ima_visualization_and_differencing.py

```diff
@@ -0,0 +1,327 @@
+    ax = plt.gca()
+    for spine in ['top', 'bottom', 'left', 'right']: ax.spines[spine].set_visible(False)
+    plt.grid()
+    plt.xlabel('SAMPTIME [s]')
+    plt.ylabel('$\mu$ [e-/s]')
+    plt.legend(loc = 0)
+    plt.title(ima)
+        
+        
+def panel_plot(cube, integ_time, median_diff_full_frame, median_diff_lhs, median_diff_rhs,
+               standard_dev_fullframe, standard_dev_lhs, standard_dev_rhs, diff_method):
+    '''
+    Plot the difference between individual reads of an IMA image file in a
+    panel plot up to 4x4 in size (i.e. SCI[16]-SCI[15], SCI[15]-SCI[14], SCI[14]-SCI[15], etc.).
+    
+    Parameters
+    ----------
+
+    ima: str
+        Name of the IMA file.
+    integ_time : array-like
+        Integration times associated with the datacube in ascending order.
+
+    median_diff_full_frame: array-like
+        The median difference in signal between each full-frame read.
+
+    median_diff_lhs: array-like
+        The median difference in signal between the left side of each read.
+
+    median_diff_rhs: array-like
+        The median difference in signal between the right side of each read.
+     
+    standard_dev_fullframe : array of floats
+        The standard deviation of the median signal of the full frame of each read.
+        
+    standard_dev_lhs : array of floats
+        The standard deviation of the median signal of the left side of each read.
+        
+    standard_dev_rhs : array of floats
+        The standard deviation of the median signal of the right side of each read. 
+        
+    diff_method: str
+        The method of finding the difference between reads. 
+        Either "instantaneous" or "cumulative".
+    
+    Returns:
+    --------
+    
+    fig: figure object
+        Panel plot with subplots showing the difference between subsequent IMA reads. 
+        Above each panel, we print the mean difference $\mu$ in the count rate over the entire image. 
+        Below each panel, we list the IMSET difference, along with the time interval between the two IMSETs.
+        The statistics in orange (on the left and right side of each panel) give the mean rate and 
+        standard deviation of each side of the image, respectively. The value in green 'delta' is the 
+        difference between the left and right side of the image. 
+        The value in white "Ratio" gives the ratio of the mean difference in orange 
+        for the left versus the right side. 
+    '''
+    
+    
+    xlabel_list = ["SCI[16-15]","SCI[15-14]","SCI[14-13]","SCI[13-12]","SCI[12-11]",
+                 "SCI[11-10]","SCI[10-9]","SCI[9-8]","SCI[8-7]","SCI[[7-6]]","SCI[6-5]",
+                "SCI[5-4]","SCI[4-3]","SCI[3-2]","SCI[2-1]"]
+    
+    fig, axarr = plt.subplots(4, 4)
+    fig.set_size_inches(40, 40)
+    fig.set_dpi(40)
+    itime = integ_time[0:-1] - integ_time[1:]
+
+    siglvl = (0.5,0.5)
+    
+    ima_j = cube[:, :, 1:]
```

Comment:
Just call `compute_diff_imas`. That way it's already defined before the if-else statement below and it already raises an error in case the method is incorrect.
```
diff = compute_diff_imas(cube, integ_time, diff_method)
```